px.py

```diff
@@ -681,8 +681,7 @@ def do_proxy_type(self):
                         proxy_auth += header[1] + " "
 
                 for auth in proxy_auth.split():
-                    auth = auth.upper()
-                    if auth in ["NTLM", "KERBEROS", "NEGOTIATE", "BASIC"]:
+                    if auth.upper() in ["NTLM", "KERBEROS", "NEGOTIATE", "BASIC"]:
                         proxy_type = auth
```

Comment:
Maybe save it as upper case so that in the check below, it is already upper and doesn't need to be done in a loop.

Reply:
I had saved it upper-cased before but reasoned that 4 iterations isn't that much of an impact and looked more legible :) Have a look at commit [777a7284f569b98cdf7892833b1f9c10f3d9c107](https://github.com/genotrance/px/pull/107/commits/777a7284f569b98cdf7892833b1f9c10f3d9c107). 

If you're ok with what I had before then I can revert that commit as well. just let me know

Reply:
Ah, never mind, that's the whole problem - storing it as upper case is wrong. We should store the auth as received like you are and only check in upper case.